### PR TITLE
Gfm tables

### PR DESCRIPTION
**Pull request by narphorium**

### CODE REVIEW

File: claat/render/md.go [hunk 1]
```diff
@@ -323,21 +324,46 @@ func (mw *mdWriter) youtube(n *types.YouTubeNode) {
...
+	// Calculate table inner content length.
+	contentLength := 0
+	for _, row := range n.Rows {
+		for _, cell := range row {
+			var nw bytes.Buffer
+			WriteMD(&nw, mw.env, mw.format, cell.Content.Nodes...)
+			contentLength += len(bytes.Trim(nw.Bytes(), " \t"))
+		}
+	}
+	// If table content is empty, don't output the table.
```
Comment (DragonRider0o0) [inline claat/render/md.go:336]: Please use `GridNode.Empty()`.
See https://github.com/googlecodelabs/tools/blob/master/claat/types/node.go#L227-L237

File: claat/render/md.go [hunk 2]
```diff
@@ -323,21 +324,46 @@ func (mw *mdWriter) youtube(n *types.YouTubeNode) {
...
+
+			// Check cell content for newlines and replace with inline HTML if newlines are present.
+			var nw bytes.Buffer
+			WriteMD(&nw, mw.env, mw.format, cell.Content.Nodes...)
+			if bytes.ContainsRune(nw.Bytes(), '\n') {
+				for _, cn := range cell.Content.Nodes {
+					cn.MutateBlock(false) // don't treat content as a new block
+					var nw2 bytes.Buffer
+					WriteHTML(&nw2, mw.env, mw.format, cn)
+					mw.writeBytes(bytes.ReplaceAll(nw2.Bytes(), []byte("\n"), []byte("")))
```
Comment (DragonRider0o0) [inline claat/render/md.go:357]: Consider replacing new lines with `<br>` or ` ` rather than just removing them. <br> is probably the best path forward.

**Review by DragonRider0o0 — COMMENTED** [review]
See my comments

- https://github.com/googlecodelabs/tools/pull/506/files#r502498143

- https://github.com/googlecodelabs/tools/pull/506/files#r502500878

File: claat/render/md.go (hunk 2 above)
Comment (shawnbuso) [reply claat/render/md.go:357]: I think in this case it's newlines in the markdown source that we're replacing, so we probably don't want <br />.

e.g. we are changing this:
<ul>
  <li>thing</li>
</ul>

to this:
<ul><li>thing</li></ul>

If the goal is to preserve newlines in text, we should probably add something to the text render method that replaces them with <br />, but leave this as-is.

Comment (DragonRider0o0) [reply claat/render/md.go:357]: Shawn is right, replacing `\n` with `<br>` would be better placed in text rendering logic.

Comment (shawnbuso) [reply claat/render/md.go:357]: I think we can do that in another PR - but Shawn should address the other comment above before merging.

**Review by shawnbuso — APPROVED** [review]

**Pull request merged**